auditbeat/module/file_integrity/file_parsers_fips_test.go

```diff
@@ -0,0 +1,64 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build requirefips
```

Comment:
Is there way to accomplish the variant testing without utilizing build tags? Minimizing the amount of code behind build tags makes maintenance easier. Build tags can hide errors.

For example, is there some runtime method of checking if the binary is in FIPS mode (akin to https://pkg.go.dev/crypto/fips140#Enabled) that we can use to skip tests at runtime instead of using build tags?

Reply:
`fips140.Enabled()` returns true if `fips140` is set to on, or only.
On enables FIPS compliant algorithms, and will allow you to use non-FIPS algorithms (such as SHA1)
Only forces you to use FIPS compliance, without using non-FIPS algorithms.

The next step in our FIPS testing will be to run these unit tests with `fips140=only` just to make sure we don't accidentally add non compliant algorithms in the future.

Reply:
I didn't mean to use `fips140.Enabled()` directly, but was asking if there was something similar in nature that we could use to minimize the amount code hidden behind build tags? One possible example would be to have a very slim package that exports a constant based on the build tag used, e.g.

```
//go:build requirefips

const FIPS = true
```


```
//go:build !requirefips

const FIPS = false
```

Then utilize this value to control the expectations set by the tests.

Reply:
I'll create an issue to discuss this; enabling test coverage is our priority at the moment